### PR TITLE
Minor performance fixes

### PR DESCRIPTION
**Pull request by arj03**
Minor tweaks that improves the performance of https://github.com/pietgeursen/flume-tests

### CODE REVIEW

**Comment by mixmix** [comment]
this appears logically consistent to me

File: blocks.js [hunk 1]
```diff
@@ -35,7 +35,10 @@ module.exports = function (file, block_size, cache) {
         var cb = cbs[i]
         cbs[i] = null
         if(!err) cache.set(i, buf)
-        while(cb.length) cb.shift()(err, buf, bytes_read)
+        var l = cb.length
+        for (var j = 0; j < cb.length; ++j)
+          cb[j](err, buf, bytes_read)
+        cb = cb.slice(l)
```
Comment (mixmix) [inline blocks.js]: this line leaves `cb` in the same state but I'm not sure it's needed ... `cb` was created in this function scope and doesn't get any further use?

Comment (arj03) [reply blocks.js]: Right, cbs[i] is nulled. Good catch. Thanks!

**Comment by mixmix** [comment]
did this get published, and did you open new pr's to downstream modules @arj03 ?

**Comment by arj03** [comment]
I published this as a minor release so it should get picked up by downstream.

**Pull request merged**